docs/en/middleware/zenoh.md

```diff
@@ -0,0 +1,194 @@
+# Zenoh (PX4 ROS 2 rmw_zenoh)
+
+<Badge type="tip" text="main (planned for: PX4 v1.17)" /> <Badge type="warning" text="Experimental" />
+
+:::warning Experimental
+At the time of writing, PX4 Zenoh-pico is experimental, and hence subject to change.
+:::
+
+PX4 supports Zenoh as an alternative mechanism (to DDS) for bridging uORB topics to [ROS 2](../ros2/user_guide.md) (via the ROS 2 [`rmw_zenoh`](https://github.com/ros2/rmw_zenoh) middleware).
+This allows uORB messages to be published and subscribed on a companion computer as though they were ROS 2 topics.
+It provides a fast and lightweight way to connect PX4 to ROS 2, making it easier for applications to access vehicle telemetry and send control commands.
+
+The following guide describes the architecture and various options for setting up the Zenoh client and router.
+In particular, it covers the options that are most important to PX4 users exploring Zenoh as an alternative communication layer for ROS 2.
```

Comment:
FYI I rewrote this because PX4 docs are PX4 centric - talking about ROS2 and rmw_zenoh felt a little weird so I've made it more clear about the setup.